packages/hardhat-core/src/internal/core/tasks/util.ts

```diff
@@ -0,0 +1,16 @@
+import { TaskIdentifier } from "../../../types";
+
+export function parseTaskIdentifier(taskIdentifier: TaskIdentifier): {
+  scope: string | undefined;
+  task: string;
+} {
+  let scope: string | undefined;
```

Comment:
This may be easier to read if we do:
```
  if (typeof taskIdentifier === "string") {
    return { scope: undefined, task: taskIdentifier };
  } else {
    const { scope, task } = taskIdentifier;
    return { scope, task };
  }
```